pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -939,31 +934,39 @@ func (ctrl *PersistentVolumeController) updateVolumePhaseWithEvent(ctx context.C
 
 // assignDefaultStorageClass updates the claim storage class if there is any, the claim is updated to the API server.
 // Ignores claims that already have a storage class.
-// TODO: if resync is ever changed to a larger period, we might need to change how we set the default class on existing unbound claims
```

Comment:
I believe this TODO still applies, or did the resync period change recently? If so, can you share relevant links and PRs please?

Reply:
You're welcome. It's not easy to understand this code well, so don't get discouraged :) 

In this case the TODO item was basically referring to a potential problem with this approach if the resync time ever increases, but the periodic resync itself is problematic and there were intentions for refactoring (no idea what the current status is): https://github.com/kubernetes/kubernetes/issues/85661